src/Controller/EnMarche/CitizenProjectController.php

```diff
@@ -45,4 +49,34 @@ public function helpAction(): Response
 
         return new Response();
     }
+
+    /**
+     * @Route("/skills/autocompletion",
+     *     name="app_citizen_project_skills_autocomplete",
+     *     condition="request.isXmlHttpRequest() and request.query.get("category") and request.query.get("term")"
```

Comment:
It's gonna do a 404 if we missed `category` and `term` parameters from the query string ? If yes it would be better to answer with a 400

Reply:
Yeah I'm pretty cool with this since this is internal api call that should not happen in the first place unless we want a full list to start but then it should be part of the first request unless the list is lazy loaded because the filed is optional. Anyway of to add some `if` in the controller as long as we don't trigger services for nothing.

Reply:
Agree thank you ;-)